### DISCUSSION TITLE
Ability to truncate coordinates when aggregated

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5575] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
While you can truncate the coordinates in the drop downs and QCBXs in how they display, you cannot do this for the aggregated output.

![image](https://user-images.githubusercontent.com/37256050/164517238-2248954b-7c7f-4e25-8bfd-79367711eb7b.png)

**Describe the solution you'd like**
A simple way to truncate coordinates in the aggregation settings.

**Reported By**
CSIRO

**Comment by maxpatiiuk** [comment]
This actually might be fixable by adding an attribute like `initialize="step=0.0001"` to the LatLongUI plugin definition.

Documentaiton: https://github.com/specify/specify7/wiki/Form-System#latlonui

Then, in your formatter definition, use the `lat1text` and `long1text` fields instead of `locality1` and `longitude1`

The change to the `step` attribute would be applied to new and modified localities only

**Comment by maxpatiiuk** [comment]
I see this in my DataObjFormatters.xml:

```xml
<format
        name="Locality"
        title="Locality"
        class="edu.ku.brc.specify.datamodel.Locality"
        default="true"
        >
        <switch single="true">
            <fields>
                <field>localityName</field>
                <field sep="; ">geography.fullName</field>
                <field type="bigdecimal" format="%9.2f" sep="; ">latitude1</field>
                <field type="bigdecimal" format="%9.2f" sep=", ">longitude1</field>
            </fields>
        </switch>
    </format>
```

Also, I see that sp7 back-end code looks at that `format` attribute.

Could you check if adding that attribute works?
Please check several places:
1. Formatting a locality record in query results (resource is formatted by the back-end)
2. Opening a form that included a query combo box to locality (resource is formatted by the front-end)

**Comment by maxpatiiuk** [comment]
@grantfitzsimmons does the above formatter work in sp7?

it should be supported in places where formatting is done by back-end (query results, autocomplete suggestion in query combo boxes), but not when formatting is done by front-end (query combo box value after page reload)

**Comment by grantfitzsimmons** [comment]
There is an issue opened specifically because this does not work correctly:

https://github.com/specify/specify7/issues/2351

**Comment by maxpatiiuk** [comment]
Ah, I forgot that the other one was created.
**Should we close one of the two?** Seems like this is an effect and the other one is a cause - no point in keeping them separate
If you agree, please merge them into one as you see fit

**Comment by grantfitzsimmons** [comment]
@maxpatiiuk in #2351 

> See DataObjFormatter definition:
> 
> ```xml
>       <format
>         name="Locality"
>         title="Locality"
>         class="edu.ku.brc.specify.datamodel.Locality"
>         default="true"
>         >
>         <switch single="true">
>             <fields>
>                 <field>localityName</field>
>                 <field sep="; ">geography.fullName</field>
>                 <field type="bigdecimal" format="%9.2f" sep="; ">latitude1</field>
>                 <field type="bigdecimal" format="%9.2f" sep=", ">longitude1</field>
>             </fields>
>         </switch>
>     </format>
> ```
> 
> The `format` attribute on `<field />` is ignored by the front-end, but causes the back-end to get confused and output the raw value of the output rather than a formatted output:
> 
> ![Screen Shot 2022-10-25 at 12 03 58](https://user-images.githubusercontent.com/40512816/197837313-490fda8a-3d6a-4a20-bba9-1d63dc587581.png)
> 
> Seems to happen here - https://github.com/specify/specify7/blob/774d108124c8d8f74c4e24feead2def47a83e5eb/specifyweb/stored_queries/format.py#L67
> 
> Discovered by @grantfitzsimmons
